src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/UserPropertyImpl.java

```diff
@@ -55,6 +58,30 @@ public String getDisplayName() {
         public UserProperty newInstance(User user) {
             return null;
         }
+
+        @SuppressWarnings("unused")
+        public FormValidation doCheckAuthorizedKeys(@QueryParameter String authorizedKeys, @AncestorInPath User currentUser) {
+            for (User user: User.getAll()) {
+                UserPropertyImpl userPropertyImpl = user.getProperty(UserPropertyImpl.class);
+                if (!authorizedKeys.isEmpty() && userPropertyImpl!=null && userPropertyImpl.authorizedKeys!=null && user!=currentUser && checkReapeatedAuthorizedKeys(authorizedKeys, userPropertyImpl)) {
+                    return FormValidation.error("There is at least one user with the same SSH public Key");
+                }
+            }
+            return FormValidation.ok();
+        }
+
+        private boolean checkReapeatedAuthorizedKeys(String userAuthorizedKeys, UserPropertyImpl userPropertyImpl) {
```

Comment:
I prefer to unify the implementation with `#isAuthorizedKey` to make sure the check is still up to date.